docs/adr/001-file-metadata-mariadb.md

```diff
@@ -0,0 +1,90 @@
+# 📝 Architecture Decision Record (ADR)
+
+## Title  
+Storing File Metadata and Extracted Entities in MariaDB
+
+## Status  
+**Proposed**
+
+## Context  
+We need to store structured metadata extracted from PDF documents uploaded to `research.digitalni-muzeum.cz`.  
+Each file is associated with a project (i.e., a folder), has a format and flags like OCR or public visibility, and will later be linked to a user.  
+In addition to file-level metadata, the system must store **extracted entities** such as persons, locations, and years – including their **type**, **value**, and **location within the document** (e.g., page number).  
+These records must be efficiently queryable, as the system is optimized for **read-heavy workloads**.
+
+## Decision  
+Use **MariaDB** and implement a normalized schema with the following tables:
+
+- `research_files`: stores file-level metadata
+- `research_entities`: stores extracted entities from each file
+- `research_users`: stores user information -- NEXT PHASE
+- `research_roles`: stores user role definitions -- NEXT PHASE
+
+This design supports structured querying, future user/project associations, and clear separation of concerns.
+
+## Alternatives Considered
+
+| Option | Pros | Cons |
+|--------|------|------|
+| MariaDB with normalized tables | Compatible with existing stack, efficient indexing, clear structure | Requires joins and schema evolution for new needs |
+| JSON document store (e.g., MongoDB) | Good for unstructured/nested data | Inconsistent with existing systems, harder for structured queries |
+| CSV / flat file metadata | Easy initial setup | Poor scalability, hard to search, no relationships |
```

Comment:
I think we could simply serialize the data into a JSON and store it to a separate file. It would be even easier to set it up (native serializers are available) and would be easier to search. In fact, it would be a fine choice for a PoC until we decide on a DB we would use.

CSV or flat files don't make much sense here.